### ISSUE TITLE
Problem with column "SampleName" in results.csv

### ISSUE THREAD

**Issue by ejlaqua** [terrimporter/MetaWorks#14]
Hello! I’m having a problem with the column “SampleName” not working in the results.csv file after running MetaWorks.

- When I ran the pipeline (with the Default ESV Workflow) the results.csv “SampleName” column would contain sample names in the format “S1_PrimerName”, “S2_PrimerName”… corresponding to the {sample} wildcards from the input *.fastq.gz files plus the primer name from the adapters_anchored.fasta file.
- But now when I run it all samples are named “results_PrimerName” or “MetaWorks1_PrimerName” and there’s no way for me to see or organize the data from the different samples. 
- I’ve noticed this happening in results.csv while using the Default ESV Workflow with the testing data, multiple classifiers including 16S_vertebrates, COI and 12S_vertebrates and with MetaWorks1.12.0 and with MetaWorks1.13.0 . Attached is an example of when “SampleNames” was not working.

I'm having difficulty troubleshooting this and can't figure out if something is wrong with my file names or somewhere else. Any help or advice would be appreciated!

![Screenshot 2024-04-10 104257](https://github.com/terrimporter/MetaWorks/assets/143235511/63dba6ae-97b8-49a6-80ff-8928b26bf016)

**Comment by terrimporter** [comment]
Hi,

I think maybe your sequence file naming format may have changed or you may need to edit the file name pattern in the config file.

Does it help if you adjust the pattern?

Terri


From: ejlaqua ***@***.***>
Date: Wednesday, April 10, 2024 at 3:00 PM
To: terrimporter/MetaWorks ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [terrimporter/MetaWorks] Problem with column "SampleName" in results.csv (Issue #14)

Hello! I’m having a problem with the column “SampleName” not working in the results.csv file after running MetaWorks.

  *   When I ran the pipeline with MetaWorks1.12.0 (Default ESV Workflow) the results.csv “SampleName” column would contain sample names in the format “S1_PrimerName”, “S2_PrimerName”… corresponding to the {sample} wildcards from the input *.fastq.gz files plus the primer name from the adapters_anchored.fasta file.
  *   But now when I run it using MetaWorks1.13.0 all samples are named “results_PrimerName” or “MetaWorks1_PrimerName” and there’s no way for me to see or organize the data from the different samples.
  *   I’ve noticed this problem in results.csv with the Default ESV Workflow, the testing data, and using multiple classifiers including 16S_vertebrates and 12S_vertebrates. Attached is an example of when “SampleNames” was not working.

I'm having difficulty troubleshooting this and any help or advice would be appreciated!

Screenshot.2024-04-10.104257.png (view on web)<https://github.com/terrimporter/MetaWorks/assets/143235511/63dba6ae-97b8-49a6-80ff-8928b26bf016>

**Comment by ejlaqua** [comment]
Hi Terri! I was able to solve the issue! It turns out that it was caused by the paths. I had included the full path in this wildcard section of the "config_ESV.yaml" file. I believe that the dots in the "/MetaWorks1.13.0" part of the path caused one of the SED expressions in one of the scripts to pull out "MetaWorks1" as the sample name for all of the samples instead of the correct sample name.

![Screenshot 2024-04-17 144046](https://github.com/terrimporter/MetaWorks/assets/143235511/3eb86d1e-9fa3-4bed-9cbc-447b36a1456c)

Including the full path like this wasn't necessary! When I replaced it with this shorter path the "SampleNames" error went away.

![Screenshot 2024-04-17 144013](https://github.com/terrimporter/MetaWorks/assets/143235511/de1da213-3c52-4ef0-bbe6-d434269a439d)

I'll be more careful about the paths in the future to avoid this type of error! Thanks!
